allennlp/tests/data/dataset_readers/coreference/coref_reader_test.py

```diff
@@ -184,6 +185,81 @@ def test_read_from_file(self, lazy):
         assert (["they"], 1) in gold_mentions_with_ids
         assert (["the", "clever", "Hong", "Kong", "people"], 1) in gold_mentions_with_ids
 
+    def test_wordpiece_modeling(self):
+        tokenizer = PretrainedTransformerTokenizer("bert-base-cased")
+        conll_reader = ConllCorefReader(
+            max_span_width=self.span_width, wordpiece_modeling_tokenizer=tokenizer
+        )
+        instances = ensure_list(
+            conll_reader.read(str(AllenNlpTestCase.FIXTURES_ROOT / "coref" / "coref.gold_conll"))
+        )
+
+        assert len(instances) == 3
+
+        fields = instances[2].fields
+        text = [x.text for x in fields["text"].tokens]
+
+        assert text == [
+            "[CLS]",
+            "Hong",
+            "Kong",
+            "Wet",
+            "##land",
+            "Park",
+            ",",
+            "which",
+            "is",
+            "currently",
+            "under",
+            "construction",
+            ",",
+            "is",
+            "also",
+            "one",
+            "of",
+            "the",
+            "designated",
+            "new",
+            "projects",
+            "of",
+            "the",
+            "Hong",
+            "Kong",
+            "SA",
+            "##R",
+            "government",
+            "for",
+            "advancing",
+            "the",
+            "Hong",
+            "Kong",
+            "tourism",
+            "industry",
+            ".",
+            "[SEP]",
+        ]
+
+        spans = fields["spans"].field_list
+        span_starts, span_ends = zip(*[(field.span_start, field.span_end) for field in spans])
+
+        candidate_mentions = self.check_candidate_mentions_are_well_defined(
+            span_starts, span_ends, text
+        )
+
+        # Asserts special tokens aren't included in the spans
+        assert all(span_start > 0 for span_start in span_starts)
+        assert all(span_end < len(text) - 1 for span_end in span_ends)
+
+        gold_span_labels = fields["span_labels"]
+        gold_indices_with_ids = [(i, x) for i, x in enumerate(gold_span_labels.labels) if x != -1]
+        gold_mentions_with_ids: List[Tuple[List[str], int]] = [
+            (candidate_mentions[i], x) for i, x in gold_indices_with_ids
+        ]
+
+        assert (["Hong", "Kong"], 0) in gold_mentions_with_ids
+        # Within span_width before wordpiece splitting but exceeds afterwards
+        assert (["the", "Hong", "Kong", "SA", "##R", "government"], 0) not in gold_mentions_with_ids
```

Comment:
This is a fine assertion to include too, but my point from before is that we should have a positive test case here, e.g.

```
assert (["foo", "##bar"], 0) in gold_mentions_with_ids
```

As written one can't tell whether the code is actually including any mentions that were split which is fundamentally what this change is for.